src/playwrightTestServer.ts

```diff
@@ -255,11 +255,12 @@ export class PlaywrightTestServer {
       if (!locations && !testIds)
         return;
 
-      const result = await this._runGlobalHooksInServer(debugTestServer, 'setup', reporter);
+      const result = await Promise.race([
+        this._runGlobalHooksInServer(debugTestServer, 'setup', reporter),
```

Comment:
I think we should pass the cancellation token to `_runGlobalHooksInServer()` instead, and make sure to pass it from any other callsites.